go/services/reports/pkg/filter/application_logs_filter.go

```diff
@@ -0,0 +1,26 @@
+package filter
+
+import (
+	"github.com/Magpie-Monitor/magpie-monitor/pkg/repositories"
+	"github.com/Magpie-Monitor/magpie-monitor/services/reports/pkg/insights"
+)
+
+func FilterByApplicationsAccuracy(
+	logsByApplication map[string][]*repositories.ApplicationLogsDocument,
+	configurationByApplication map[string]*insights.ApplicationInsightConfiguration,
+) {
+
+	for application, logs := range logsByApplication {
+		// Check if application configuration is in params
+		config, ok := configurationByApplication[application]
+		var accuracy insights.Accuracy
+		if !ok {
+			// By default the app is not included
+			delete(logsByApplication, application)
+		} else {
+			accuracy = config.Accuracy
+			filter := NewAccuracyFilter[*repositories.ApplicationLogsDocument](accuracy)
```

Comment:
Can we refactor accuracy filter so that it can be created once and filter based on accuracy passed to .Filter method?

Reply:
Not sure about this solution as it would make AccuracyFilter itself less generic.